### PR TITLE
Reject instance create requests for invalid names

### PR DESCRIPTION
**Pull request by mandoonandy**
Resolves shakenfist/terraform-provider-shakenfist#12

### CODE REVIEW

**Comment by mandoonandy** [comment]
I meant to add that this is my *proposed* solution to TF issue 12.

**Comment by mikalstill** [comment]
I feel a bit like we should solve this in SF as described in my comment on the issue. We should talk about this before merging.

File: shakenfist/external_api/app.py [hunk 1]
```diff
@@ -607,8 +607,9 @@ def post(self, name=None, cpus=None, memory=None, network=None,
              instance_uuid=None):
         global SCHEDULER
 
-        # We need to sanitise the name so its safe for DNS
-        name = re.sub(r'([^a-zA-Z0-9_\-])', '', name)
+        # Check that the instance name is safe for use as a DNS host name
+        if name != re.sub(r'([^a-zA-Z0-9_\-])', '', name) or len(name) > 63:
+            return error(400, "Instance name must be useable as a DNS host name")
```
Comment (mikalstill) [inline shakenfist/external_api/app.py:612]: Quotes! I see quotes!

Comment (mandoonandy) [reply shakenfist/external_api/app.py:618]: Rebased to current master and fixed the quotes

**Pull request merged**